### DISCUSSION TITLE
Hits widget and useHits have different behaviour

### DISCUSSION THREAD

**Discussion by terencere** [algolia/instantsearch#6013] [Q&A]
Here I approach two ways to show hits:
(1). Use Hits widget

```
<Index indexName="computers">
        <Hits hitComponent={Results} />
</Index>
<Index indexName="phones">
        <Hits hitComponent={Results} />
</Index>
```

(2). use useHits hooks

```
function CustomResults() {
    const { hits, results, sendEvent } = useHits();
    return (
      {hits.map((h) => (
        <Results hit={h}  />
      ))}
    )
}
```

then

```
<Index indexName="computers">
        <CustomResults />
</Index>
<Index indexName="phones">
        <CustomResults />
</Index>
```

As above, when use Hits widget can show both computers and phones, but when use useHits hook it can only show phones (actually it just show the last index results).
Why these two ways show different results, how to make (2). works like (1).

**Comment by Haroenv** [comment]
There's no difference between that implementation using useHits and Hits. Are you sure that's the only change? Do you have a sandbox with this behaviour showing?
